kr-core/src/main/clojure/edu/ucdenver/ccp/kr/sparql.clj

```diff
@@ -243,11 +244,43 @@
        (sparql-query-body optional-clauses)
        " }\n "))
 
+(defn filter-body [filter-clauses]
+  (str " FILTER ( "
+       (operator-body filter-clauses)
+       " )\n "))
+
+(defn bind-body [bind-clauses]
+  (str " BIND "
+       (operator-body bind-clauses)
+       " \n "))
+
+(defn iri-body [iri-clauses]
+  (str " IRI "
+       (operator-body iri-clauses)
+       " \n "))
+
+(defn concat-body [concat-clauses]
+  (str " CONCAT "
+       (operator-body concat-clauses)
+       " \n "))
+
+(defn strafter-body [strafter-clauses]
+  (str " STRAFTER "
+       (operator-body strafter-clauses)
+       " \n "))
+
+
+
+
 ;; (def operator-bound [args]
 ;;      (str " bound(" (filter-body (first args))  ") "))
 
 (declare operator-body)
 
+(defn naked-unary-operator [op-name]
+  (fn [args]
+    (str " " op-name (operator-body (first args)))))
+
```

Comment:
`naked-unary-operator` and `unary-operator` could be collapsed into a single function that takes a keyword argument `:naked?` (which defaults to `false`) to indicate whether the parens are needed.  

Either way, it would be nice to have some documentation on the function that describes when each form is appropriate.